test/gurka/test_speed_smoothing.cc

```diff
@@ -0,0 +1,211 @@
+#include "gurka.h"
+#include "test.h"
+
+using namespace valhalla;
+
+class SmallRouteSpeedTest : public ::testing::Test {
+protected:
+  static void SetUpTestSuite() {
+    const std::string ascii_map = R"(
+      A----B
+    )";
+
+    const gurka::ways ways = {
+        {"AB", {{"highway", "primary"}}},
+    };
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, 10000);
+    map = gurka::buildtiles(layout, ways, {}, {}, "test/data/small_route_speeds",
+                            {
+                                {"mjolnir.shortcuts", "false"},
+                                {"mjolnir.timezone", VALHALLA_BUILD_DIR "test/data/tz.sqlite"},
+                            });
+    map.config.put("mjolnir.traffic_extract", "test/data/small_route_speeds/traffic.tar");
+    map.config.put("thor.invariant_recosting", false);
+
+    // add live traffic
+    test::build_live_traffic_data(map.config);
+    current_ = 52;
+    freeflow_ = 100;
+    constrained_ = 40;
+    historical_max_ = 31;
+    historical_min_ = 19;
+    test::customize_live_traffic_data(map.config, [&](baldr::GraphReader&, baldr::TrafficTile&, int,
+                                                      valhalla::baldr::TrafficSpeed* traffic_speed) {
+      traffic_speed->overall_encoded_speed = current_ >> 1;
+      traffic_speed->encoded_speed1 = current_ >> 1;
+      traffic_speed->breakpoint1 = 255;
+    });
+
+    test::customize_historical_traffic(map.config, [&](DirectedEdge& e) {
+      e.set_constrained_flow_speed(constrained_);
+      e.set_free_flow_speed(freeflow_);
+
+      // speeds for every 5 min bucket of the week
+      std::array<float, kBucketsPerWeek> historical;
+      for (size_t i = 0; i < historical.size(); ++i) {
+        size_t min_timestamp = (i % (24 * 12)) / 12;
+        if (min_timestamp < 12) {
+          historical[i] = historical_max_ - min_timestamp;
+        } else {
+          historical[i] = historical_min_ + (min_timestamp - 12);
+        }
+      }
+      return historical;
+    });
+  }
+
+  static gurka::map map;
+  static uint32_t current_, historical_max_, historical_min_, constrained_, freeflow_;
+};
+
+gurka::map SmallRouteSpeedTest::map = {};
+uint32_t SmallRouteSpeedTest::current_ = 0, SmallRouteSpeedTest::historical_max_ = 0,
+         SmallRouteSpeedTest::historical_min_ = 0, SmallRouteSpeedTest::constrained_ = 0,
+         SmallRouteSpeedTest::freeflow_ = 0;
+
+TEST_F(SmallRouteSpeedTest, LiveTrafficSpeed) {
+  auto reader = test::make_clean_graphreader(map.config.get_child("mjolnir"));
+  for (auto tile_id : reader->GetTileSet()) {
+    auto tile = reader->GetGraphTile(tile_id);
+    for (const auto& e : tile->GetDirectedEdges()) {
+      auto speed = tile->GetSpeed(&e, baldr::kCurrentFlowMask, 0);
+      EXPECT_EQ(speed, current_);
+
+      speed = tile->GetSpeed(&e, baldr::kCurrentFlowMask, 10000);
+      EXPECT_EQ(speed, current_);
```

Comment:
could you add more comments what you check ? (I see comments only for some `EXPECT_EQ` checks)

Reply:
I moved the tests to test_traffic_routes file. Added comments everywhere